tests/unit/accounts/test_views.py

```diff
@@ -418,6 +418,109 @@ def test_request_password_reset(
             pretend.call(pyramid_request, stub_user),
         ]
 
+    def test_request_password_reset_with_email(
+            self, monkeypatch, pyramid_request, pyramid_config, user_service,
+            token_service):
+
+        stub_user = pretend.stub(email=pretend.stub())
+        pyramid_request.method = "POST"
+        token_service.dumps = pretend.call_recorder(lambda a: "TOK")
+        user_service.get_user_by_username = pretend.call_recorder(
+            lambda a: None
+        )
+        user_service.get_user_by_email = pretend.call_recorder(
+            lambda a: stub_user
+        )
+        pyramid_request.find_service = pretend.call_recorder(
+            lambda *a, **kw: user_service,
+        )
```

Comment:
You probably want something like this here: https://github.com/pypa/warehouse/blob/5783a2678f6a36ed5fbd5427ed112887e0baff1a/tests/unit/accounts/test_views.py#L438-L443

Reply:
Yes, I figured I needed to change that, but came upa less elegant solution:
``` python
pyramid_request.find_service = pretend.call_recorder(
            lambda *a, **kw:
            token_service if 'name' in kw else user_service,
        )
```
I'm debating as to overriding the `max_age` attribute of the `token_service` with a simple `3600` and not a stub. Any opinions?

Reply:
Making it an integer makes sense to me, since the view is now doing actual math with it.

Reply:
It wasn't required after all /shrugs